support cfb_shell and cfb samples on native_posix/native_posix_64

soburi opened the pull request:
This PR makes cfb_shell and cfb samples work on native_posix/native_posix_64.

- sdl_dc (and some displays) enable display_blanking on boot up. Turn it off on sample startup or `cfb init` command at shell.
- cfb_shell not set the display pixel format. Set the pixel format to MONO10 or MONO01.

jfischer-no commented on subsys/fb/cfb.c:
```diff
@@ -330,6 +330,7 @@ int cfb_framebuffer_init(const struct device *dev)
 	struct display_capabilities cfg;
 
 	api->get_capabilities(dev, &cfg);
+	api->blanking_off(dev);
```
This is not necessary, it is up to the app to disable it, see https://github.com/zephyrproject-rtos/zephyr/blob/1c8f1cd590e10dcc349ebd8649c313293e29209c/samples/subsys/display/cfb/src/main.c#L40

soburi replied:
Thak you for reviewing.

I misunderstood some design policies.
I rewritten this patch to modify only the sample.

jfischer-no commented on samples/subsys/display/cfb_shell/boards/native_posix.conf:
```diff
@@ -0,0 +1 @@
+CONFIG_SDL_DISPLAY_DEFAULT_PIXEL_FORMAT_MONO01=y
```
This is doubly problematic, we should always try to avoid board specific overlays because it adds maintenance overhead and it is not the correct way, it should be done by the app, something like https://github.com/zephyrproject-rtos/zephyr/blob/1c8f1cd590e10dcc349ebd8649c313293e29209c/samples/subsys/display/cfb/src/main.c#L26-L29 should be added to https://github.com/zephyrproject-rtos/zephyr/blob/11a2107d991b14a7abb7081498c914a9f57ed089/subsys/fb/cfb_shell.c#L410-L429

soburi replied:
It' make a sense. I fixed it.

soburi committed:
samples: display: cfb_custom_font: turn off display_blanking at boot

Turn off display_blanking to make it able to refresh on a device
even when blanking default off.

Signed-off-by: TOKITA Hiroshi <tokita.hiroshi@fujitsu.com>

soburi committed:
fb: cfb_shell: turn off display_blanking at initializing

Turn off display_blanking to make it able to refresh on a device
even when blanking default off.

Signed-off-by: TOKITA Hiroshi <tokita.hiroshi@fujitsu.com>

soburi committed:
fb: cfb_shell: set pixel format as MONO10 or MONO01 on initializing

Set the display's pixel format to MONO10 or MONO01 that cfb supports
on cfb initializing.

Signed-off-by: TOKITA Hiroshi <tokita.hiroshi@fujitsu.com>
